internal/testutil/statespace/statespace.go

```diff
@@ -0,0 +1,98 @@
+package statespace
+
+import (
+	"math/rand/v2"
+	"strings"
+	"testing"
+)
+
+// Model implements a simple fuzz-style test framework.
+// Rather than fuzz scalar values like property-based tests, this framework enumerates the subsets of a bounded, high-cardinality state space.
+// Essentially, the framework will call the subject with every subset of the known mutations and assert on each invariant.
+// Application order of mutations is randomized to avoid order-dependence without fully permuting the space (the implementation is very fast).
```

Comment:
Do these comments make any sense? 🤣 

We're not _quite_ doing property-based testing but I'm not aware of a better term.